Fix up links so they are 508 compliant

TomNUSDS opened the pull request:
Doing this at the jekyll doesn't work because github's build system limits the plugins a site can use.

Another solution is to do it in javascript.

Image showing matched and unmatched links:
<img width="720" alt="image" src="https://github.com/usds/website/assets/74203452/c1f4d89f-5668-4bf6-9ffb-bd0efbda407a">


One possible issue are buttons that are links that take users offsite.
<img width="238" alt="image" src="https://github.com/usds/website/assets/74203452/31489015-9cc3-49f3-afc2-25d358c9f019">

TomNUSDS committed:
Fix up links so they are 508 compliant in js

Doing this at the jekyll doesn't work because github's build system limits the plugins a site can use.

drew-usds commented:
@TomNUSDS 

In general I'm seeing the script add the `usa-link--external` class, the `rel=` and the `target=` to external links 🎉  👍 .

---

### In addition to your flag on buttons, found the following things doing QA (adding them here as a to-do for me to check off)

- [x] Some _internal_ links are being treated as external when viewing locally (imagine the same would happen on staging). Could present problems for QA both locally and on staging. Obvi solution is to just go clean those up, but could be an annoying ongoing QA thing .

> Ex: the lead grafs on`/apply` and the nine-year blog post have a hardcoded link to `https://usds.gov/*` that is being treated as external when viewing locally/on staging. 
>![Screenshot 2024-04-23 at 5 11 30 PM](https://github.com/usds/website/assets/57154955/4af88e81-0bb5-4729-b348-1b3d9c73f655)
>
> ![Screenshot 2024-04-23 at 5 03 37 PM](https://github.com/usds/website/assets/57154955/cc0967ae-4f87-4bae-8072-84f0d51732f9)

- [x] Spacing after the links in body text is a little tight, could be expanded

> ![Screenshot 2024-04-23 at 5 13 52 PM](https://github.com/usds/website/assets/57154955/52da1a4f-ef21-432e-9ff2-c4abb3d29703)


- [x] Social link icons are getting the style added to them, which overlaps with the icon.

> ![Screenshot 2024-04-23 at 5 09 30 PM](https://github.com/usds/website/assets/57154955/2bcf4020-427a-469a-bdc7-48793ea2ac18)
> ![Screenshot 2024-04-23 at 4 40 18 PM](https://github.com/usds/website/assets/57154955/c82e89b7-bca7-4ece-927d-651d792b5bba)

- [x] Baseline of external links with smaller text (ex in press clips, in the footer) don't _totally_ line up with the external icon. Looks like it might be the external link icon margin being tied to the x height (`.7ex`) of the larger body font. _This is pretty minimal, though!_

> ![Screenshot 2024-04-23 at 4 40 13 PM](https://github.com/usds/website/assets/57154955/5cdaec82-ec0c-4595-983e-b26b506f72c5)
> ![Screenshot 2024-04-23 at 5 07 59 PM](https://github.com/usds/website/assets/57154955/0c68c25d-70b8-48dd-a2db-c8f44d27bb7f)

```
.usa-link--external.usa-link--alt::after{
...
margin-top: .7ex;
}
```

---

### Other thoughts

- [ ] Not sure how this will behave with the `/apply` form. Won't be able to QA until we push it live.

TomNUSDS commented:
> Some internal links are being treated as external when viewing locally

I think this is just a local thing, but I'll see if I can't make it work.

> Spacing after the links in body text is a little tight, could be expanded

I think this is just standard USWDS layout for the style. We can tweak.

> Social link icons are getting the style added to them, which overlaps with the icon.

This is a straight up bug. I fixed it for the first attempt (using the Jekyll plugins), but forgot to do it here.

> Baseline of external links with smaller text (ex in press clips, in the footer) don't totally line up with the external icon.

I may need help with the css here. the `.usa-link--external.usa-link--alt::after` may make the other link locations too high. It may need to be specifically targeted at elements under the list element?

TomNUSDS committed:
Fixes

- Ignore www.usds.gov and usds.gov absolute links (impacts local dev and staging server)
- Do not add icon to social links (but still target="_blank", etc)
- Added .scss selectors to control positions better. Attempted to fix, but @drew-usds you should feel free to make changes to this branch to tweak.

TomNUSDS committed:
Update externallinks.js

drew-usds reviewed (approved):
👍
